.github/workflows/csharp_databricks_e2e.yml

```diff
@@ -0,0 +1,141 @@
+      - 'csharp/src/Drivers/Apache/Spark/**'
+      - 'csharp/src/Drivers/Databricks/**'
+      - 'csharp/test/Drivers/Databricks/**'
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.head_ref || github.sha }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+  id-token: write  # Required for OIDC token exchange
+
+defaults:
+  run:
+    # 'bash' will expand to -eo pipefail
+    shell: bash
+
+jobs:
+  csharp-databricks-e2e:
+    name: "C# ${{ matrix.os }} ${{ matrix.dotnet }}"
+    runs-on: ${{ matrix.os }}
+    environment: databricks-e2e
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        dotnet: ['8.0.x']
+        os: [ubuntu-latest, windows-2022, macos-13, macos-latest]
+    steps:
+      - name: Install C#
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - name: Checkout ADBC
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          fetch-depth: 0
+          submodules: recursive
+      - name: Build
+        shell: bash
+        run: ci/scripts/csharp_build.sh $(pwd)
+      - name: Set up Databricks testing
+        shell: bash
+        env:
+          DATABRICKS_WORKSPACE_URL: 'adb-6436897454825492.12.azuredatabricks.net'
+          DATABRICKS_WAREHOUSE_PATH: '/sql/1.0/warehouses/2f03dd43e35e2aa0'
+          DATABRICKS_SP_CLIENT_ID: '8335020c-9ba9-4821-92bb-0e8657759cda'
+        run: |
+          # Set up cross-platform variables
+          if [[ "$RUNNER_OS" == "Windows" ]]; then
+            DATABRICKS_DIR="$USERPROFILE/.databricks"
+            DATABRICKS_CONFIG_FILE="$USERPROFILE/.databricks/connection.json"
+          else
+            DATABRICKS_DIR="$HOME/.databricks"
+            DATABRICKS_CONFIG_FILE="$HOME/.databricks/connection.json"
+          fi
+
+          # Get GitHub OIDC token
+          GITHUB_TOKEN=$(curl -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" \
+            "$ACTIONS_ID_TOKEN_REQUEST_URL&audience=https://github.com/apache" | jq -r '.value')
+
+          if [ "$GITHUB_TOKEN" = "null" ] || [ -z "$GITHUB_TOKEN" ]; then
+            echo "Failed to get GitHub OIDC token"
+            exit 1
+          fi
+
+          # Exchange OIDC token for Databricks OAuth token
+          OAUTH_RESPONSE=$(curl -X POST "https://$DATABRICKS_WORKSPACE_URL/oidc/v1/token" \
+            -H "Content-Type: application/x-www-form-urlencoded" \
+            -d "grant_type=urn:ietf:params:oauth:grant-type:token-exchange" \
+            -d "client_id=$DATABRICKS_SP_CLIENT_ID" \
+            -d "subject_token=$GITHUB_TOKEN" \
+            -d "subject_token_type=urn:ietf:params:oauth:token-type:jwt" \
+            -d "scope=sql")
+
+          DATABRICKS_TOKEN=$(echo "$OAUTH_RESPONSE" | jq -r '.access_token')
```

Comment:
You might want to use [`::add-mask::`](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-commands#masking-a-value-in-a-log) in some areas here to ensure that the access tokens don't show up in the logs